controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/RoutingPolicies.java

```diff
@@ -192,17 +192,22 @@ private void removeObsoleteEndpointsFromDns(LoadBalancers loadBalancers, Deploym
                    .collect(Collectors.toSet());
     }
 
-    /** Load balancers for a particular deployment */
-    private static class LoadBalancers {
+    /** Load balancers allocated to a deployment */
+    private static class AllocatedLoadBalancers {
 
         private final ApplicationId application;
         private final ZoneId zone;
         private final List<LoadBalancer> list;
 
-        private LoadBalancers(ApplicationId application, ZoneId zone, List<LoadBalancer> list) {
+        private AllocatedLoadBalancers(ApplicationId application, ZoneId zone, List<LoadBalancer> loadBalancers) {
             this.application = application;
             this.zone = zone;
-            this.list = list;
+            this.list = loadBalancers.stream()
+                                     // DNS no longer needs to maintain for load balancer in other states
+                                     // (e.g. inactive) than the following
+                                     .filter(lb -> lb.state() == LoadBalancer.State.active ||
+                                                   lb.state() == LoadBalancer.State.reserved)
```

Comment:
Semantics of this filter statement is unclear to me. Maybe extract the predicate to a separate method where the name is more explanatory?

Reply:
Done.